apps/yggdrasil/priv/repo/migrations/20160229164207_populate_roles.exs

```diff
@@ -0,0 +1,40 @@
+defmodule Yggdrasil.Repo.Migrations.PopulateRoles do
+  use Ecto.Migration
+
+  def up do
+    execute """
```

Comment:
So having been forced to implement a RoleResourcePermission object, this could be done in Elixir via the seed script I think.

For now I'm leaving it. The reason for all the common table expressions is that apparently postgress does not allow variables in the query like T-SQL so either temp tables or the CTEs are suggested so the form below from what I have read is idiomatic.

Just looks odd. If get time I may try to do in the seed file.